boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig

```diff
@@ -25,3 +25,11 @@ CONFIG_UART_CONSOLE=y
 
 # additional board options
 CONFIG_GPIO_AS_PINRESET=y
+
+# Specific to beadedstream Spot Logger and D605N
+# (might want to make a separate board file at some point)
+
+# LFCLK/SYSTEM CLOCK/32.768Khz clock source
+CONFIG_CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING=y
```

Comment:
That's very funny :)
I'm pretty sure I was setting up this parameter but don't remember if I had a can of spray at that moment. Need to retest with Kyler

Why not to simply pass it as a CMake parameter in upload_bootloader.sh?

Reply:
Because it's a hardware KConfig parameter.    You're probably circumventing a bunch of error checks if you pass it on the make command line, and I'm not sure that would work properly.   You want it in a board config file or a .conf file of some sort --- this comes from Zephyr's documentation that I read last night.

This is hardware setting specific to the board.   It should really be in only one place, in the board file(s), because anything you write for either the spot logger or the D605N should be setting the LFCLK source the same way (including the bootloader.)      I originally had added it to the app board files for the spot logger and D605N also, but then I saw someone had already added it to the prj.conf file.  It's a hardware setting, so that's not where I would have put it... but hey, whatever.   😆

Reply:
As far as I understand you're referring to this:

<img width="546" alt="image" src="https://github.com/beadedstream/zephyr/assets/109213907/6c763644-1e91-4b03-b27d-bbd5c7dcfe3f">

I agree with you, that's definitely where this parameter must be located. But that also means we need to go to the D605N  repo and refactor the way how we setting up the board configuration. The problem is that we're limited in resources (only 2 people are working on the project) and that doesn't have the highest priority now. This is what I planned to do later and why asked to pass this parameter as a cache entry for now.

>the bootloader spins waiting for the LFCLK to switch from RC to external oscillator

Yes, that's true but not always. Sometimes the device is passed through that function and just operates "normally" until you will try to connect to BLE. But then you will got a disconnect immediately. Nordic's Softdevice depends on 32kHz as well. That could explain why I could experience some issues with connection. The problem is that it's not clear how Nordic setups the clocking. 

By default `CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL` is chosen (not `CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC`). So before this change the device was operating as it use a regular XTAL (without a bypass feature being enabled):

<img width="728" alt="image" src="https://github.com/beadedstream/zephyr/assets/109213907/cab6c431-320e-4dcf-be61-2492638d2d98">

The LFCLK register value (`NRFX_CLOCK_CONFIG_LF_SRC`) is chosen based on Zephyr configurations: `CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC`, `CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL`, `CONFIG_CLOCK_CONTROL_NRF_K32SRC_EXT_LOW_SWING` and so on in `nrfx_config.h` file. So by default it was equal to `1` (instead of `196609` when setting up `CONFIG_CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING`).

And it was operating just normally until we cooled down the device. What changed internally in the nRF52840 chip so the BLE stopped working normally or device got stuck at spinwait function because the corresponding interrupts were not called? This is what is not cleared for me. 

Looks like some of the internal elements (probably in a nRF52840 chip) significantly changed their physical properties so the cpu decided that no proper crystal was installed.

Reply:
So don't make any changes to the D605N build/config scheme.   I'm just telling you how I would have done it, I didn't say you had to do it that way.   I would caution against _not_ passing Kconfig parameters to the build via "-DCONFIG_etc_etc" on the command line, as I'm not sure that will work right.   When you do it the recommended way, I _think_ there may be some dependency and error checking that goes on (not sure about that, but it looked like it, especially when I made a typo and specified an option that did not exist.

If you look in the build directory, there's a .config file that gets generated.   I _think_ that's the configuration the build system decides on after it combines the board and project conf files.   After I fixed the bootloader (via the zephyr fork), that file has the FULL_SWING constant for the LFCLK source.   

I mentioned to Rodrigo that I know I fixed at least one problem ...  and I verified the bootloader itself no longer hangs on startup.   There may be other issues compounding the same problem.   I asked him to let me know if after testing, he wanted me to look at the problem more.